modules/querying/pages/query-operations.adoc

```diff
@@ -1142,3 +1142,101 @@ The scope of the command depends on the user's current scope.
 If the user has set a working graph, then `DROP ALL` removes all the queries for that graph.
 If a user has set their scope to be global, then `DROP ALL` removes all queries across all graph spaces.
 
+== Query Descriptions
+
+Query descriptions explains the procedures for updating, showing, and dropping descriptions of queries and query parameters.
+
+=== Update description
+After creating a query, you can add a description for the query.
+
+*Syntax*
+[source]
+----
+UPDATE DESCRIPTION OF <objectType> <objectName> [ON GRAPH <graphName>] <description>
+----
+
+Required privilege
+`WRITE_QUERY`
+
+To update a description, run the `UPDATE DESCRIPTION` command as shown below. You can update the description of a query or a query parameter. If you choose not to specify a graph in the command, the current scope will be used as the scope of the description.
+
+[source, gsql]
+----
+GSQL > USE GRAPH POC_GRAPH
+GSQL > UPDATE DESCRIPTION OF QUERY query1 "xxx"
+GSQL > UPDATE DESCRIPTION OF QUERY query2 ON GRAPH graph1 "xxx"
+GSQL > UPDATE DESCRIPTION OF QUERY_PARAM query1.param1 "xxx"
+----
+
+[NOTE]
+====
+The term `UPDATE` here means adding a description to the target.
+If the target does not already have a description, it will be added.
+If the target already has a description, it will be replaced with the new one.
+====
+
+[CAUTION]
+====
+Using create or replace is preferable for updating a query to ensure that the description is retained.
+Otherwise, there is a risk of losing the description.
+====
+
+=== Show description
+
+*Syntax*
+[source]
+----
+SHOW DESCRIPTION OF <objectType> <objectName> [ON GRAPH <graphName>]
+----
+
+Required privilege
+`READ_QUERY`
+
+To display a description, use the `SHOW DESCRIPTION` command as shown below. Similar to the update command, you can display the description of a query or a query parameter.
+
+
+[source, gsql]
+----
+GSQL > USE GRAPH POC_GRAPH
+GSQL > SHOW DESCRIPTION OF QUERY query1
+GSQL > SHOW DESCRIPTION OF QUERY query1,query2
```

Comment:
This example shows that is is possible to have a comma-separated list of queries or parameters. The syntax above does not indicate that, so technically, the syntax description is wrong, or at least overly simplified.

A simply way to do this would be to replace <objectName> with <objectName(s)>

and below the syntax, add
"where `<objectName(s)>` can be a comma-separated list of names.

We can do this sort of natural language commentary with this "informal" syntax. We could not do it with EBNF, which has very strict rules. ENBF is supposed to be understood by a computer program, e.g. a syntax checker could read your EBNF rules and your code, and then tell you if you code follows the EBNF rules.